milli/src/update/index_documents/transform.rs

```diff
@@ -546,20 +546,28 @@ impl<'a, 'i> Transform<'a, 'i> {
         })
     }
 
-    /// Returns a `TransformOutput` with a file that contains the documents of the index
-    /// with the attributes reordered accordingly to the `FieldsIdsMap` given as argument.
+    /// Clear all databases. Returns a `TransformOutput` with a file that contains the documents
+    /// of the index with the attributes reordered accordingly to the `FieldsIdsMap` given as argument.
+    ///
     // TODO this can be done in parallel by using the rayon `ThreadPool`.
-    pub fn remap_index_documents(
+    pub fn prepare_for_documents_reindexing(
         self,
-        wtxn: &mut heed::RwTxn,
+        wtxn: &mut heed::RwTxn<'i, '_>,
```

Comment:
```suggestion
        wtxn: &mut heed::RwTxn,
```

I don't think we need these lifetimes?

Reply:
I think we do. I struggled to remove them. Rust is too dumb to elide them...

Reply:
yea, smart enough to suggest them, not enough to elide them